lua/orgmode/org/autocompletion/omni.lua

```diff
@@ -101,7 +110,26 @@ local function omni(findstart, base)
     return -1
   end
 
-  local fetcher_ctx = { base = base, line = line }
+  -- The function omni gets called twice when used by omnicompletion but when
```

Comment:
I don't think we should have any completion specific logic here. I would rather introduce something like `Url.from_omni(line, base)` and figure out the URL from there.

Also, you can just do a `gsub` without checking, it will just return the string if there is no match. This will avoid one regex call:

```lua
local url_str = base:gsub('^file:', '')
```